crates/api/src/delegation/contract_delegation.rs

```diff
@@ -0,0 +1,78 @@
+use std::sync::Arc;
+
+use alloy_provider::{
+    fillers::{BlobGasFiller, ChainIdFiller, FillProvider, GasFiller, JoinFill, NonceFiller},
+    network::Ethereum,
+    Identity, RootProvider,
+};
+use alloy_sol_types::sol;
+use alloy_transport::BoxTransport;
+use ethereum_consensus::primitives::{BlsPublicKey, BlsSignature};
+use helix_common::api::constraints_api::{PreconferElection, SignedPreconferElection};
+use reth_primitives::{Address, Bytes};
+use TaiyiCore::TaiyiCoreInstance;
+
+use super::{error::Error, traits::DelegationTrait};
+
+const DEFAULT_GAS_LIMIT: u64 = 100_000;
+
+type RecommendProvider = FillProvider<
+    JoinFill<Identity, JoinFill<GasFiller, JoinFill<BlobGasFiller, JoinFill<NonceFiller, ChainIdFiller>>>>,
+    RootProvider<BoxTransport>,
+    BoxTransport,
+    Ethereum,
+>;
+sol! {
+    #[derive(Debug)]
+    struct PreconferElectionRes {
+        bytes validatorPubkey;
+        bytes preconferPubkey;
+        uint256 chainId;
+        address preconferAddress;
```

Comment:
What is preconferAddress used for? maybe not needed?

Reply:
Just want to keep it aligned with https://github.com/lu-bann/taiyi/blob/b3dc7487e3c6b7e75b5223bcd36155f93cf74ea3/contracts/src/interfaces/IDelegationContract.sol#L36-L41. I should put comments here .